model/src/main/java/org/projectnessie/model/IcebergTable.java

```diff
@@ -69,20 +63,19 @@
   @NotBlank
   public abstract String getMetadataLocation();
 
-  /** Opaque representation of Iceberg's {@code TableIdGenerators}. */
-  public abstract String getIdGenerators();
+  public abstract String getTableState();
```

Comment:
I thnk we should put actual properties for each thing here rather than an opaque `String` (same comment as @dimas-b  I think).

Off the top of my head I think we should save as actual properties:
* partition spec id
* schema id
* sequence number
but I may be missing some from V2 spec.

Reply:
@rymurr: sequence number is a global field. So no need to keep in the on_reference_state.
we just have to keep fields that are specific to current branch.

So, I think current chosen parameters are ok  (If table property need to maintain for per branch, we can add that as well)

And agree on the part that we can have a class called `TableCurrentState`, which keeps these parameters instead of opaque string.

Reply:
> we just have to keep fields that are specific to current branch.

ok cool. agreed

>  sequence number is a global field.

I would just like to amend this statement a bit for my own clarity. We have to save anything that can be changed w/o creating a snapshot. The real risk we face is there are iceberg transactions that don't create a snapshot. So global counters (column id, sequence number) are safe to ignore. Things like schema and partition id are not.

Reply:
Additionally I think we should create a test suite (probably along w/ the iceberg fix) to test all these assumtions